File: Data Structures & Algorithms/maximum-product-subarray/submission-0.cs
public class Solution {
    public int MaxProduct(int[] nums) {
        int res = 0;
        int curMin = 1;
        int curMax = 1;

        foreach (int num in nums){
            int temp = num * curMax;
            curMax = Math.Max(Math.Max(num*curMax, num*curMin),num);
            curMin = Math.Min(Math.Min(temp, num*curMin), num);
            res = Math.Max(res, curMax);
        }
        return res; 
        
    }
}

File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, nums: List[int]) -> int:
        left = 0
        right = len(nums)-1

        max_area = 0

        while left != right:
            area = min(nums[left],nums[right]) * (right - left)
            max_area = max(max_area,area)
            left += 1

        return max_area   
        